Updates the PX4 Architecture diagrams.

hamishwillee opened the pull request:
This updates the Dronecode Platform architecture doc. There is no "dronecode platform" in the way that was presented. The change splits the image into two cases - with FC and FC+ mission computer. It attempts to only present things that are "supported" - ie it isn't aspirational.

@bkueng Can you please review. If you are broadly happy with it I will further
- Link up as many bits as I can to existing docs
- rename the actual source file.

FYI @LorenzMeier

bkueng commented on en/concept/dronecode_architecture.md:
```diff
@@ -1,17 +1,53 @@
+
+The hardware consists of 
+- [Flight controller](../flight_controller/README.md) (running the PX4 flight stack). This often includes internal IMUs, compass and barometer.
+- [Motor ESCs](../peripherals/esc_motors.md) connected to [PWM outputs](../peripherals/pwm_escs_and_servo.md) or a [UAVCAN](../peripherals/uavcan_escs.md) bus (UAVCAN allows two-way communication, not single direction as shwown).
```
I'd keep this a bit broader as there's no only the 2 protocols, or motors: Actuators (motors and servos).
Also on the diagram

hamishwillee replied:
OK, so add Dshot and also say "or other protocols"?

bkueng replied:
Yes, but there's also more buses, like I2C or serial.

hamishwillee replied:
Ah, I'd only seen ESC connected to PWM or UAVCAN - had assumed I2C or serial were never used for this

hamishwillee replied:
Fixed.

bkueng commented on en/concept/dronecode_architecture.md:
```diff
@@ -1,17 +1,53 @@
+
+The diagram below shows a PX4 system that includes both a flight controller and a mission computer.
+
+![PX4 architecture - FC only system](../../assets/diagrams/px4_arch_fc_companion.svg)
```
```suggestion
![PX4 architecture - FC and Mission Computer](../../assets/diagrams/px4_arch_fc_companion.svg)
```

bkueng reviewed (commented):
Yes looks good

hamishwillee commented:
Thanks for review. Not sure why it is failing, but I will check , fix issue, and merge (probably tomorrow)

hamishwillee commented:
I've fixed up following feedback, moved the page, cross linked to relevant resources. Now merging. @bkueng I think this is good, but feel free to review again!
